api/server/router/system/system_routes.go

```diff
@@ -44,6 +47,28 @@ func (s *systemRouter) pingHandler(ctx context.Context, w http.ResponseWriter, r
 		w.Header().Set("Content-Length", "0")
 		return nil
 	}
+
+	if capabilitiesQueryValue := r.URL.Query().Get("capabilities"); capabilitiesQueryValue != "" {
+		wantedCapabilitiesVersion, err := strconv.Atoi(capabilitiesQueryValue)
+		if err != nil {
+			return errdefs.InvalidParameter(err)
+		}
+		capabilities, err := s.handleCapabilities(ctx, wantedCapabilitiesVersion)
+		if err != nil {
+			log.G(ctx).WithError(err).Error("failed")
```

Comment:
I need to jump out in a bit, but wondering if this should still fall through  to the plain-text "OK" response, or of this should return an error (will have to look closer)

Reply:
Yeah, I was also wondering about that. I probably lean on that side, I feel like the client is specifically requesting capabilities, so if we fail the call should fail. OTOH, this seems less likely to cause problems, people put comments in daemon.json `features`, who knows if someone out there is putting random query params in their pings 😅

Reply:
I went looking in what situations this would return an error, and it looks like currently there are none? (see other comment up/down (don't know where github will put it) 😂 

Somewhat considering that if things went wrong, it probably shouldn't ignore it. Or if we do, it could return an empty capabilities response (JSON, empty struct, or response with empty capabilities)